commit 08d64453e5675e6d7f0022091d544a54b831a0a0
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Mar 17 00:48:23 2017 +0000

    PCI: Add comments about ROM BAR updating
    
    [ Upstream commit 0b457dde3cf8b7c76a60f8e960f21bbd4abdc416 ]
    
    pci_update_resource() updates a hardware BAR so its address matches the
    kernel's struct resource UNLESS it's a disabled ROM BAR.  We only update
    those when we enable the ROM.
    
    It's not obvious from the code why ROM BARs should be handled specially.
    Apparently there are Matrox devices with defective ROM BARs that read as
    zero when disabled.  That means that if pci_enable_rom() reads the disabled
    BAR, sets PCI_ROM_ADDRESS_ENABLE (without re-inserting the address), and
    writes it back, it would enable the ROM at address zero.
    
    Add comments and references to explain why we can't make the code look more
    rational.
    
    The code changes are from 755528c860b0 ("Ignore disabled ROM resources at
    setup") and 8085ce084c0f ("[PATCH] Fix PCI ROM mapping").
    
    Link: https://lkml.org/lkml/2005/8/30/138
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Reviewed-by: Gavin Shan <gwshan@linux.vnet.ibm.com>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1425_08d64453e5675e6d7f0022091d544a54b831a0a0.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1425_08d64453e5675e6d7f0022091d544a54b831a0a0.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
